### PR TITLE
add doc for spack install with external modules

### PR DESCRIPTION
**Pull request by gknapp1**
I've added documentation for installing the `additivefoam` spack module when you are using modules external to spack. This is useful when you need to install OpenFOAM on the system in addition to the AdditiveFOAM solver and you have system modules available to you for common libraries like C++ and MPI.

### CODE REVIEW

File: docs/installation.md [hunk 1]
```diff
@@ -19,6 +19,57 @@ spack install additivefoam
...
 
+### Spack installation with external modules
+
+If you are installing OpenFOAM through spack, some configuration is needed to enable spack to use external modules, which is common on HPC systems. For OpenFOAM, this is largely needed for the C++ compiler and MPI dependencies, which are both generally available as preconfigured modules that can be loaded for functionality that has been configured for a specific HPC system. While module versions and names may vary from system to system, this example provides a template for configuring spack on your system.
+
+First, the desired system compiler needs to be added to the spack compiler configuration.
+
+```shell
+module load gcc/8.1.0
+spack compiler add $(dirname $(which gcc))
```
Comment (streeve) [inline docs/installation.md:30]: Does this actually depend on the module setup? Couldn't you do this anywhere?

File: docs/installation.md [hunk 2]
```diff
@@ -19,6 +19,57 @@ spack install additivefoam
...
+```
+
+Second, you must get the location prefix of the OpenMPI installation by copying the output of the below command.
```
Comment (streeve) [inline docs/installation.md:33]: Any MPI?

File: docs/installation.md [hunk 3]
```diff
@@ -19,6 +19,57 @@ spack install additivefoam
 ```
 spack `develop` is currently required.
 
+### Spack installation with external modules
+
+If you are installing OpenFOAM through spack, some configuration is needed to enable spack to use external modules, which is common on HPC systems. For OpenFOAM, this is largely needed for the C++ compiler and MPI dependencies, which are both generally available as preconfigured modules that can be loaded for functionality that has been configured for a specific HPC system. While module versions and names may vary from system to system, this example provides a template for configuring spack on your system.
```
Comment (streeve) [inline docs/installation.md:24]: I would revise to say this example will use a standard HPC module setup, but the general idea of using an existing compiler/MPI can work anywhere

**Review by streeve — COMMENTED** [review]
I could be wrong, but I think this is more general: "spack install with existing compiler/dependencies"

**Comment by gknapp1** [comment]
I addressed the above comments. It's a good point that this could be more general. One of the other use cases that I didn't think of originally is that if there is a system module for OpenFOAM-10, then you can create a spack package specification to wrap it and still use the spack installation of additivefoam.

I'd guess it will be less straightforward to use a standalone install of OpenFOAM-10 with spack, becuase of the environmental setup that would need to be run when loading the spack module. But I don't know enough about spack to write an example of that right now.

**Review by streeve — APPROVED** [review]

**Comment by streeve** [comment]
@colemanjs good to merge?

**Pull request closed without merging**